CMakeLists.txt

```diff
@@ -7,24 +7,56 @@ SET(CMAKE_CXX_EXTENSIONS OFF)
 
 PROJECT(h2olog)
 
-SET(CXX_WARNING_FLAGS "-Wall")
+IF(NOT CMAKE_BUILD_TYPE)
+    SET(CMAKE_BUILD_TYPE Release)
+ENDIF(NOT CMAKE_BUILD_TYPE)
 
-SET(CMAKE_CXX_FLAGS "${CXX_WARNING_FLAGS} ${CMAKE_CXX_FLAGS}")
+SET(CMAKE_CXX_FLAGS_RELEASE  "-O3 -flto")
+
+SET(CMAKE_CXX_FLAGS "-Wall -Wextra -Wno-unused-parameter -g3 ${CMAKE_CXX_FLAGS}")
 
 FIND_PATH(LIBBCC_INCLUDE_DIR bcc/compat)
 FIND_LIBRARY(LIBBCC_LIB "bcc")
 IF (LIBBCC_LIB)
   IF (LIBBCC_INCLUDE_DIR)
+    INCLUDE_DIRECTORIES("${LIBBCC_INCLUDE_DIR}")
     INCLUDE_DIRECTORIES("${LIBBCC_INCLUDE_DIR}/bcc/compat")
-    SET(LIBBCC_LIBRARIES "-lbcc")
   ELSE (LIBBCC_INCLUDE_DIR)
     MESSAGE(SEND_ERROR "libbcc header files not found")
   ENDIF (LIBBCC_INCLUDE_DIR)
 ELSE (LIBBCC_LIB)
   MESSAGE(SEND_ERROR "libbcc not found")
 ENDIF (LIBBCC_LIB)
 
-SET(SOURCE_FILES h2olog.cc http.cc quic.cc)
+SET(SOURCE_FILES
```

Comment:
I saw that you introduced a new file: data-types.h -- Does that need to go in here too? I'd also  love to know your thoughts on the future of the header file. Should all other common definitions live there too? if that's the case, I prefer H2O's convention of using the word "common". If not, quic.h might be nice. Thanks!

Reply:
8c7168abcf823d6a545589a3c7fc91ebacf9a497

`data-types.h` has partial copies of structs from h2o and quicly. "data-types-from-h2o-and-quicly.h" might be correct, I guess.

Reply:
Or probably we can use `h2o.h` and `quicly.h` directly, and then declare fields we use in an allow-list in `gen-bpf.py`.